src/main/java/net/earthcomputer/clientcommands/command/PingCommand.java

```diff
@@ -49,23 +48,16 @@ private static int printPing(FabricClientCommandSource source, Collection<GamePr
             throw PLAYER_IN_SINGLEPLAYER_EXCEPTION.create();
         }
 
-        if (profiles.isEmpty()) {
+        if (profiles.size() != 1) {
             throw PLAYER_NOT_FOUND_EXCEPTION.create();
         }
-        if (profiles.size() > 1) {
-            throw TOO_MANY_PLAYERS_EXCEPTION.create();
-        }
```

Comment:
I think it's better to keep these as separate exceptions for UX reasons. If you target multiple players it doesn't make sense to say "no player found".

Reply:
It's nonsensical to query more than one game profile, I'd even say that the argument type returning a `Collection` is a quirk.

Reply:
Does it not return multiple values if you use a target selector?

Reply:
It does, but for the purposes of this command (and every other command using this argument type), it does not make sense to want that. If the query does result in more than one profile, it makes sense to say that _the_ player was not found, based on the target selector given by user who expected a single result. I suppose this is also the reason Mojang does not use this argument type internally anymore at all, and now uses `EntityArgument.player()`.

Reply:
Just because it's nonsensical to do doesn't mean there shouldn't be a proper error message for it.

Reply:
So you're saying we should add a `TOO_MANY_PLAYERS_EXCEPTION` everywhere where we use this argument type?

Reply:
Yes. If that becomes a problem we can add a helper method but afaik this is the only place we use it.

Reply:
We use it in lots of places, and importantly without this extra exception.

Reply:
I have replaced every usage of `CGameProfileArgument.gameProfile()` with `CEntityArgument.player()`. This allows for the same functionality but does not allow multiple matches when an entity selector is used.

Reply:
It actually turns out the `profiles.isEmpty()` check we did everywhere was unnecessary, that is already prevented by the argument type 😄.